api/lib/routes/aggregated_metrics.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+module.exports = function(settings) {
+  var express = require("express");
+  var router = express.Router();
+  var logger = require("../log/logger");
+  var HttpStatus = require("http-status-codes");
+  var eventGeneratorUtil = require("../utils/eventGeneratorUtils")(settings.eventGenerator);
+  var metricHelper = require("./metricHelper");
+  var routeHelper = require("./routeHelper");
+
+  router.get("/:app_id/aggregated_metric_histories/:metric_type", function(req, resp) {
+    var appId = req.params.app_id;
+    var metricType = req.params.metric_type;
+    var startTime = req.query["start-time"];
+    var endTime = req.query["end-time"];
```

Comment:
when end-time is missing,  API server set it to "-1":
https://github.com/cloudfoundry-incubator/app-autoscaler/blob/develop/api/lib/routes/metricHelper.js#L36-L38

Then,  when retrieving metrics from appmetricdb or instancemetricdb,  the implementation ensures that the default value is Now():
https://github.com/cloudfoundry-incubator/app-autoscaler/blob/develop/src/autoscaler/db/sqldb/appmetric_sqldb.go#L111-L113

Anyway, of course we can change the metricHelper.js to represent Now() directly.

Reply:
@boyang9527 ,  I would like to keep the implementation of https://github.com/cloudfoundry-incubator/app-autoscaler/blob/develop/api/lib/routes/metricHelper.js#L36-L38  in API server now instead of converting the end-time to Date.now(). 
The reason is that Date.now() return a timestamp in milliseconds,  but in backend go component,  we uses nanoseconds .. 
If we pass the timestamp in milliseconds as end-time to eventGen, then the query result should be wrong.  
If we change milliseconds to nanoseconds by appending additional "000000" ,  it seems not right to expose such a detail Go implementation in API server side. 
The current implementation looks more safe  ..